pyprep/find_noisy_channels.py

```diff
@@ -545,9 +554,11 @@ def run_ransac(
         n_chns = chn_pos.shape[0]
 
         # Before running, make sure we have enough memory
+        max_ram_usage = 0.95
         try:
             available_gb = virtual_memory().available * 1e-9
             needed_gb = (data[:n_chns, :].nbytes * 1e-9) * n_samples
+            needed_gb = needed_gb / max_ram_usage  # adjust for max use
             assert available_gb > needed_gb
```

Comment:
I agree with the point you and @yjmantilla make here, but this is not very readable.

the point is: from overall available RAM, reduce a portion because we anticipate a user to be needed some of that RAM after analyses are started

so I would expect you to tweak `available_gb`, not `needed_gb` (also, dividing with a float < 1 is confusing IMHO)

I suggest:

```suggestion
        # we anticipate users to also need RAM for other things next to pyprep
        max_to_use_from_available_ram = 0.95
        try:
            available_gb = virtual_memory().available * 1e-9
            available_gb *= max_to_use_from_available_ram
            needed_gb = (data[:n_chns, :].nbytes * 1e-9) * n_samples
            assert available_gb > needed_gb
```

Reply:
I agree with this. Is better that way for readability. Regarding the MemoryError, @a-hurst , you say it is confusing because in available_gb (if multiplied by the percentage) would not be the "real" available gbs or because of what?

Reply:
then I'd say we just adjust the error message

Reply:
I've done this in the latest commit: instead of reporting needed vs available RAM in the exception message, it now just reports the additional required RAM which should avoid confusion. Let me know what you think!

Reply:
yes, that's great - thanks!